client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/GroupedInvestigations/useGroupedInvestigations.ts

```diff
@@ -24,30 +24,56 @@ const useGroupedInvestigations = ({ invetigationsToGroup, onClose }: useGroupedI
     const { alertError } = useCustomSwal();
     
     const onSubmit = (data: GroupForm) => {
-        const groupId = uuidv4();
-        const groupToCreate = {
-            id: groupId,
-            reason: data.reason?.id,
-            otherReason: data.otherReason
-        };
-        const groupToCreateLogger = logger.setup({
-            workflow: 'create grouped investigations',
-            user: userId,
-            investigation: epidemiologyNumber
-        });
-        groupToCreateLogger.info('launching grouped investigations request', Severity.LOW);
-        const invetigationsToGroupIds = invetigationsToGroup.map((invetigationToGroup: InvestigationTableRow) => invetigationToGroup.epidemiologyNumber);
-        setIsLoading(true);
-        axios.post('/groupedInvestigations', { groupToCreate, invetigationsToGroupIds })
-        .then(() => {
-            onClose();
-            groupToCreateLogger.info('create grouped investigations successfully', Severity.LOW);
-        })
-        .catch((err) => {
-            groupToCreateLogger.error(`create grouped investigations was failde due to${err}`, Severity.HIGH);
-            alertError('לא ניתן היה לקבץ חקירות אלו');
-        })
-        .finally(() => setIsLoading(false))
+        const groupIdsWithRepeation = invetigationsToGroup.map((invetigationToGroup: InvestigationTableRow) => invetigationToGroup.groupId);
+        const groupIdsWithoutRepeation = groupIdsWithRepeation.filter((item, pos) => groupIdsWithRepeation.indexOf(item) === pos && item !== null);
+        if (groupIdsWithoutRepeation.length > 0) {
+            const group = groupIdsWithoutRepeation[0];
+            const createOrUpdate = 'Update'
+            const groupToUpdateLogger = logger.setup({
+                workflow: 'update grouped investigations',
+                user: userId,
+                investigation: epidemiologyNumber
+            });
+            groupToUpdateLogger.info('launching grouped investigations request', Severity.LOW);
+            const invetigationsToGroupIds = invetigationsToGroup.map((invetigationToGroup: InvestigationTableRow) => invetigationToGroup.epidemiologyNumber);
```

Comment:
you use `invetigationsToGroupIds` twice (you do the same action also in line 63)
move it out from the if else case